lib/perl/Genome/Config.pm

```diff
@@ -0,0 +1,90 @@
+package Genome::Config;
+
+use strict;
+use warnings;
+
+use Genome::Carp qw(croakf);
+use Genome::ConfigSpec qw();
+use Path::Class qw();
+use YAML::Syck qw();
+
+sub get {
+    my $key = shift;
+    my $spec = spec($key);
+
+    my $value = _lookup_value($spec);
+    if (!$spec->has_default_value && !defined($value)) {
+        croakf('required config key missing: %s', $key);
+    }
+
+    if (!defined($value)) {
+        return $spec->default_value;
+    }
+
+    for my $v (@{$spec->validators}) {
+        if (my $error = $v->($value, $spec)) {
```

Comment:
I don't think I've quite addressed what you're suggesting but I did make it so the spec class creates the full error message,

```
my @errors = $spec->validate($value);
if (@errors) {
  my $msg = $spec->validation_error(@errors);
  croakf($msg);
}
```